.github/workflows/test-trigger-on-push.yaml

```diff
@@ -0,0 +1,42 @@
+name: Test on Push Trigger
+
+on:
+  push:
```

Comment:
Why copy/paste all the code from [test.yaml](https://github.com/BrianPugh/install-micropython/blob/main/.github/workflows/test.yaml)?

We can just remove the
```
branches:
 -main
```
specification and delete `test-trigger-on-push.yaml`

Reply:
There are two differences with new file, the removal of the main branch requirement, and adding in the reference parameter (albeit commented out right now). Wasn't sure of your preference on these two points. Maybe reference parameter can be added to test.yaml in with a value of `v1.22.1` (latest official release)?

Reply:
what if we do something like:

```
jobs:
  test:
    strategy:
      matrix:
        mpy-reference: ["v1.22.1", ""]
    steps:
        ...
      - name: Build and Install MicroPython Unix Port
        uses: ./
        with:
          reference: ${{mpy-reference}}
```